modules/net-vpc-firewall/main.tf

```diff
@@ -27,7 +27,7 @@ locals {
         for name, rule in ruleset : {
           name                 = name
           deny                 = try(rule.deny, false)
-          rules                = try(rule.rules, [{ protocol = "all" }])
+          rules                = try(rule.rules, [{ protocol = "all", ports = null }])
```

Comment:
is this needed? I remember an ugly permadiff if we set this but that might not happen anymore... I would test with an actual apply :)

Reply:
This is used only if you don't have a `rules` block in the firewall rule. In any other case we use whatever the user specifies. I didn't get any permadiffs when applying.